### PR TITLE
Added fallback copy for when we're not able to retrieve a erc721 or erc1155 name in the setApprovalForAll screen

### LINKED ISSUE #17516: Replace fallback copy for when we're not able to retrieve a erc721 or erc1155 name in the setApprovalForAll screen
## Background 
Currently when we're not able to retrieve a name for the NFT in the setApprovalForAll screen, we use the `NFT` as the fallback copy. We should update this fallback copy to `NFTs of this collection` where this collection is a link to that contract on the block explorer (currently the `NFT` fallback is the link for it).

I've highlighted in the screenshots below the places where we currently use this fallback. 

## Screenshot

1. When giving a setApprovalForAll allowance:
Screenshot 2023-01-31 at 12 44 51

2. When revoking a setApprovalForAll allowance:
Screenshot 2023-01-31 at 12 53 00

### PR DESCRIPTION
**Pull request by VSaric**
## Explanation

Currently, on `develop`, when we're **not able** to retrieve a name for the NFT in the setApprovalForAll screen, we use the `NFT` as the fallback copy. With this PR we added fallback copy for when we're **not able** to retrieve a erc721 or erc1155 name in the setApprovalForAll screen. Now fallback copy is `NFTs of this collection` where `this collection` is a link to that contract on the block explorer (currently, on `develop`, the `NFT` fallback is the link for it).

* Fixes #17516 

#### When `giving` a setApprovalForAll allowance:

215808104-8b497c82-2c46-4e03-ad4a-5a4a8279b46e

#### When `revoking` a setApprovalForAll allowance:

215810395-2992341d-7929-4871-828f-5bdcbcd2012a

#### When `giving` a setApprovalForAll allowance (when we're **able** to retrieve a erc721 or erc1155 name):

Screenshot 2023-03-09 at 18 19 10

#### When `revoking` a setApprovalForAll allowance (when we're **able** to retrieve a erc721 or erc1155 name):

Screenshot 2023-03-09 at 18 19 37

#### When `giving` a setApprovalForAll allowance (when we're **not able** to retrieve a erc721 or erc1155 name):

Screenshot 2023-03-09 at 18 17 44

#### When `revoking` a setApprovalForAll allowance (when we're **not able** to retrieve a erc721 or erc1155 name):

Screenshot 2023-03-09 at 18 18 15

## Manual Testing Steps

1. Go to ZenAcademy (erc1155) https://etherscan.io/token/0xf64e6fb725f04042b5197e2529b84be4a925902c#writeContract
2. Click Write Contract
3. Connect MM
4. Click `SetApprovalForAll`
5. Set any operator (address) and for approved (bool) set `true` for `giving`  a setApprovalForAll allowance and `false` for `revoking` a setApprovalForAll allowance
6. Click on Write
7. See new fallback copy

### CODE REVIEW

**Comment by dzfjz** [comment]
Verified by QA

**Comment by bschorchit** [comment]
Ok, me again here 🙈 thanks for flagging a couple things I have missed before, @VSaric! ❤️

2 more small copy changes for when we're able to retrieve a name for the NFT:

1. On approved asset for both approve and revoke screens, we should display `All your <NFT>`
image

2. On revoke screen, it should read `...transfer all your <NFT>` on both title and subtitle
Screenshot 2023-03-09 at 13 17 03

File: ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.test.js [hunk 1]
```diff
@@ -281,4 +283,258 @@ describe('ConfirmApproveContent Component', () => {
...
+    expect(getByText('Parameters: false')).toBeInTheDocument();
+    expect(
+      getByText(
+        '0x095ea7b30000000000000000000000009bc5baf874d2da8d216ae9f137804184ee5afef40000000000000000000000000000000000000000000000000000000000011170',
+      ),
+    ).toBeInTheDocument();
+  });
+
+  it('should render Confirm approve page correctly and display fallback copy for when we are not able to retrieve a erc721 or erc1155 name in the setApprovalForAll screen and when revoking a setApprovalForAll allowance', () => {
+    const { queryByText, getByText, getAllByText, getByTestId, container } =
```
Comment (jpuri) [inline ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.test.js:478]: Snapshot test for specific part of the page might be better and more readable for this kind of test.

File: ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js [hunk 2]
```diff
@@ -563,7 +588,10 @@ export default class ConfirmApproveContent extends Component {
         >
           {this.renderTitle()}
         </div>
-        <div className="confirm-approve-content__description">
+        <div
+          className="confirm-approve-content__description"
+          data-testid="confirm-approve-description"
+        >
```
Comment (jpuri) [inline ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js:594]: We can also avoid adding data-testid here if we use snapshot test.

File: ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.test.js (hunk 1 above)
Comment (VSaric) [reply ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.test.js]: @jpuri Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/d7e12d4155ab6ed63a4b87f71350d9d801457c99#diff-49a399717cd27ce0bb0c752633bd4e684dd9ae6469f26f0a1dcaf607ffdc6d76).

File: ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js (hunk 2 above)
Comment (VSaric) [reply ui/pages/confirm-approve/confirm-approve-content/confirm-approve-content.component.js]: @jpuri Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/d7e12d4155ab6ed63a4b87f71350d9d801457c99#diff-49a399717cd27ce0bb0c752633bd4e684dd9ae6469f26f0a1dcaf607ffdc6d76).

File: app/_locales/en/messages.json [hunk 3]
```diff
@@ -362,9 +366,13 @@
     "message": "Approve spend limit"
   },
   "approveAllTokensTitle": {
-    "message": "Allow access to and transfer of all your $1?",
```
Comment (adonesky1) [inline app/_locales/en/messages.json:365]: I also think this of should remain, just grammatically speaking. @coreyjanssen?

File: app/_locales/en/messages.json [hunk 4]
```diff
@@ -362,9 +366,13 @@
...
+    "message": "Allow access to and transfer all your $1?",
     "description": "$1 is the symbol of the token for which the user is granting approval"
   },
+  "approveAllTokensTitleWithoutSymbol": {
+    "message": "Allow access to and transfer all your NFTs of $1?",
```
Comment (adonesky1) [inline app/_locales/en/messages.json:373]:
```suggestion
    "message": "Allow access to and transfer all of your NFTs from $1?",
```

File: app/_locales/en/messages.json [hunk 5]
```diff
@@ -381,8 +389,12 @@
   "approveTokenDescription": {
     "message": "This allows a third party to access and transfer the following NFTs without further notice until you revoke its access."
   },
+  "approveTokenDescriptionWithoutSymbol": {
+    "message": "This allows a third party to access and transfer all NFTs of $1 without further notice until you revoke its access.",
```
Comment (adonesky1) [inline app/_locales/en/messages.json:393]:
```suggestion
    "message": "This allows a third party to access and transfer all of your NFTs from $1 without further notice until you revoke its access.",
```

File: app/_locales/en/messages.json [hunk 6]
```diff
@@ -381,8 +389,12 @@
...
+    "description": "$1 is a link to contract on the block explorer when we're not able to retrieve a erc721 or erc1155 name"
+  },
   "approveTokenTitle": {
-    "message": "Allow access to and transfer of your $1?",
+    "message": "Allow access to and transfer your $1?",
```
Comment (adonesky1) [inline app/_locales/en/messages.json:397]:
```suggestion
    "message": "Allow access to and transfer of your $1?",
```

File: app/_locales/en/messages.json [hunk 7]
```diff
@@ -3132,13 +3141,21 @@
     "message": "Review your spending cap"
   },
   "revokeAllTokensTitle": {
-    "message": "Revoke permission to access and transfer all of your $1?",
+    "message": "Revoke permission to access and transfer all your $1?",
```
Comment (adonesky1) [inline app/_locales/en/messages.json:3144]:
```suggestion
    "message": "Revoke permission to access and transfer all of your $1?",
```

File: app/_locales/en/messages.json [hunk 8]
```diff
@@ -3132,13 +3141,21 @@
...
+    "message": "Revoke permission to access and transfer all your NFTs of $1?",
+    "description": "$1 is a link to contract on the block explorer when we're not able to retrieve a erc721 or erc1155 name"
+  },
   "revokeApproveForAllDescription": {
-    "message": "This revokes the permission for a third party to access and transfer all of your $1 without further notice.",
+    "message": "This revokes the permission for a third party to access and transfer all your $1 without further notice.",
     "description": "$1 is either a string or link of a given token symbol or name"
   },
+  "revokeApproveForAllDescriptionWithoutSymbol": {
+    "message": "This revokes the permission for a third party to access and transfer all your NFTs of $1 without further notice.",
```
Comment (adonesky1) [inline app/_locales/en/messages.json:3156]:
```suggestion
    "message": "This revokes the permission for a third party to access and transfer all of your NFTs from $1 without further notice.",
```

**Comment by bschorchit** [comment]
Oh my bad for the misleading recommendation. I was aiming for parity across the different places where we use similar copy in this screen. Thanks for catching those, Alex!

**Review by coreyjanssen — DISMISSED** [review]

**Comment by bschorchit** [comment]
@coreyjanssen just to confirm, you agree and approve the suggestions made by @adonesky1, right?

**Comment by coreyjanssen** [comment]
@bschorchit yep! Confirmed.

File: app/_locales/en/messages.json (hunk 8 above)
Comment (VSaric) [reply app/_locales/en/messages.json]: @adonesky1 Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/eb4ca37c7710f97deef4985ff0fe52d30ac19ddb).

File: app/_locales/en/messages.json (hunk 7 above)
Comment (VSaric) [reply app/_locales/en/messages.json]: @adonesky1 Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/eb4ca37c7710f97deef4985ff0fe52d30ac19ddb).

File: app/_locales/en/messages.json (hunk 6 above)
Comment (VSaric) [reply app/_locales/en/messages.json]: @adonesky1 Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/eb4ca37c7710f97deef4985ff0fe52d30ac19ddb).

File: app/_locales/en/messages.json (hunk 5 above)
Comment (VSaric) [reply app/_locales/en/messages.json]: @adonesky1 Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/eb4ca37c7710f97deef4985ff0fe52d30ac19ddb).

File: app/_locales/en/messages.json (hunk 4 above)
Comment (VSaric) [reply app/_locales/en/messages.json]: @adonesky1 Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/eb4ca37c7710f97deef4985ff0fe52d30ac19ddb).

File: app/_locales/en/messages.json (hunk 3 above)
Comment (VSaric) [reply app/_locales/en/messages.json:365]: @adonesky1 Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/eb4ca37c7710f97deef4985ff0fe52d30ac19ddb).

File: app/_locales/en/messages.json [hunk 9]
```diff
@@ -315,9 +315,13 @@
     "message": "Alerts"
   },
   "allOfYour": {
-    "message": "All of your $1",
+    "message": "All your $1",
     "description": "$1 is the symbol or name of the token that the user is approving spending"
   },
+  "allYourNFTsOf": {
+    "message": "All your NFTs of $1",
```
[the remark this answers is not in the document]
Comment (VSaric) [reply app/_locales/en/messages.json]: @adonesky1 Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/eb4ca37c7710f97deef4985ff0fe52d30ac19ddb).

File: app/_locales/en/messages.json [hunk 10]
```diff
@@ -315,9 +315,13 @@
     "message": "Alerts"
   },
   "allOfYour": {
-    "message": "All of your $1",
```
[the remark this answers is not in the document]
Comment (VSaric) [reply app/_locales/en/messages.json:318]: @adonesky1 Done [here](https://github.com/MetaMask/metamask-extension/pull/17992/commits/eb4ca37c7710f97deef4985ff0fe52d30ac19ddb).

**Review by adonesky1 — APPROVED** [review]
LGTM

**Comment by coreyjanssen** [comment]
I'm so sorry, but I just realized that we need to slightly rephrase this copy to make sense. @adonesky1 is it too late to make this change? Apologies!

From this:

Screenshot 2023-03-17 at 9 54 11 AM

To this: 
Allow access to and transfer of all your NFTs in this collection?

**Comment by VSaric** [comment]
> I'm so sorry, but I just realized that we need to slightly rephrase this copy to make sense. @adonesky1 is it too late to make this change? Apologies!
> 
> From this:
> 
> Screenshot 2023-03-17 at 9 54 11 AM
> 
> To this: Allow access to and transfer of all your NFTs in this collection?


Hey @coreyjanssen, I am not updated my screenshots with this latest changes. Sorry about that! This is how it looks like now with latest chenges that @adonesky1 commented:

Screenshot 2023-03-20 at 13 25 42

Is this okay to stay like this or you have some new suggestions about the tittle and tittle description?
Thank you in advance!

**Review by coreyjanssen — APPROVED** [review]

**Pull request merged**